core/src/main/java/com/linecorp/armeria/internal/http/InternalHttpData.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal.http;
+
+import com.linecorp.armeria.common.http.HttpData;
+
+/**
+ * Internal APIs of {@link HttpData}. Should not be used in user code.
+ */
+public interface InternalHttpData extends HttpData {
```

Comment:
A non-internal class like `DefaultHttpData` exposing an internal class in its type hierarchy doesn't seem like a good idea to me. I guess you did so instead of introducing `AbstractHttpData` because you didn't want to duplicate the reference counting code in `DefaultByteBufHolder`. If so, perhaps we could just give up making `ByteBufHttpData` implement `ByteBufHolder` since it's an internal type?